service/share/share.go

```diff
@@ -145,16 +146,52 @@ func (s *service) GetSharesByNamespace(ctx context.Context, root *Root, nID name
 	}
 
 	namespacedShares := make([]Share, 0)
+	nodeCh := make(chan []format.Node)
+	errCh := make(chan error)
+	wg := sync.WaitGroup{}
+	newCtx, cancel := context.WithCancel(ctx)
+
 	for _, rootCID := range rowRootCIDs {
-		nodes, err := ipld.GetLeavesByNamespace(ctx, s.dag, rootCID, nID)
-		if err != nil {
-			return nil, err
+		wg.Add(1)
+		go func(rootCID cid.Cid) {
+			nodes, err := ipld.GetLeavesByNamespace(newCtx, s.dag, rootCID, nID)
+			if err != nil {
+				errCh <- err
+				return
+			}
+			nodeCh <- nodes
+		}(rootCID)
+	}
+
+	go func() {
+		wg.Wait()
+		close(nodeCh)
+		close(errCh)
+		_ = cancel
```

Comment:
?

cancel is a func that should be called. Here you just reassign the func value to nothing.

Reply:
removed